src/views/SessionView/SessionFulfilledModal.vue

```diff
@@ -22,12 +22,25 @@ export default {
       mobileMode: "app/mobileMode"
     }),
     message() {
-      return (
-        (this.modalData.isSessionEnded
-          ? "The student has already ended their session."
-          : "Another volunteer has already joined this session.") +
-        " Thanks for trying, we really appreciate it!"
-      );
+      const thankYouMessage = "Thanks for trying, we really appreciate it!";
+      let text = "";
+
+      if (this.modalData.isSessionEnded && !this.modalData.volunteerJoined) {
+        text = "The session was canceled.";
```

Comment:
@jkeat,

Any ideas for a better message to show the user?

Reply:
**Request canceled** (if ended + no volunteer ever joined):
"The student canceled their request."

**Session ended** (if ended):
"This session has already finished."

**Request filled** (if not ended and volunteer already joined):
"Another volunteer has already joined this session."

Reply:
@alymurray 

Thank you Aly!

If possible, if a student tries to access their own canceled session again - should we say "Your request was canceled"?

Reply:
Ahh good catch. Maybe we should just combine request canceled and session ended, and just show "This session has been ended." to both students and volunteers if the session is already ended?  

Do we think it matters to the person seeing the message if the session was ended bc it was over vs. ended bc the student canceled it?

Reply:
That sounds good to me ^

One thing: in the case where...
1. Volunteer1 joins the session
2. The session ends
3. Volunteer2 tries to join

...is it better to tell Volunteer2 that the session has ended, or that another volunteer joined?

Reply:
@alymurray,

> Do we think it matters to the person seeing the message if the session was ended bc it was over vs. ended bc the student canceled it?

I think it would provide for a slightly better UX if they know exactly happened with the session, but either way, the message is still pretty understandable as to what happened

@jkeat,

Maybe saying something along the lines that the session was fulfilled by another volunteer?